src/Router.sol

```diff
@@ -58,13 +82,41 @@ contract Router is Ownable, ReentrancyGuard, IRouter {
     /// @dev Mapping of requestId to transfer receipt
     mapping(bytes32 => SwapRequestReceipt) public swapRequestReceipts;
 
+    /// @notice Ensures that only an account with the ADMIN_ROLE can execute a function.
+    modifier onlyAdmin() {
+        _checkRole(ADMIN_ROLE);
+        _;
+    }
+
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor() {
+        _disableInitializers();
```

Comment:
do we need to call `initialize` in the constructor before we do this? or what's the interaction there?